src/test/java/com/contrastsecurity/maven/plugin/sdkx/GsonFactoryTest.java

```diff
@@ -19,14 +21,59 @@ void parse_iso8601_strings_to_local_date_time() {
     final Gson gson = GsonFactory.create();
 
     // WHEN deserialize a type containing a LocalDateTime
-    final Obj obj = gson.fromJson("{\"time\": \"1955-11-12T22:04:00\"}", Obj.class);
+    final LocalDateTimeObj obj =
+        gson.fromJson("{\"time\": \"1955-11-12T22:04:00\"}", LocalDateTimeObj.class);
 
     final LocalDateTime expected = LocalDateTime.of(1955, 11, 12, 22, 4, 0);
     assertThat(obj.time).isEqualTo(expected);
   }
 
+  @Test
+  void parse_null_to_local_date_time() {
+    // GIVEN a new GSON instance that should handle java.time.LocalDateTime
+    final Gson gson = GsonFactory.create();
+
+    // WHEN deserialize a type containing a LocalDateTime that is null
+    final LocalDateTimeObj obj = gson.fromJson("{\"time\": null}", LocalDateTimeObj.class);
+
+    assertThat(obj.time).isNull();
+  }
+
+  /**
+   * Verifies that the {@code Gson} returned by the factor has requisite adapters for dealing with
+   * {@link java.time.ZonedDateTime} types.
+   */
+  @Test
+  void parse_iso8601_zoned_strings_to_zoned_date_time() {
+    // GIVEN a new GSON instance that should handle java.time.ZonedDateTime
+    final Gson gson = GsonFactory.create();
+
+    // WHEN deserialize a type containing a ZonedDateTime
+    final ZonedDateTimeObj obj =
+        gson.fromJson("{\"time\": \"1955-11-12T22:04:00.000+00:00\"}", ZonedDateTimeObj.class);
+
+    final ZonedDateTime expected = ZonedDateTime.of(1955, 11, 12, 22, 4, 0, 0, ZoneOffset.UTC);
+    assertThat(obj.time).isEqualTo(expected);
+  }
+
+  @Test
+  void parse_null_to_zoned_date_time() {
+    // GIVEN a new GSON instance that should handle java.time.ZonedDateTime
+    final Gson gson = GsonFactory.create();
+
+    // WHEN deserialize a type containing a ZonedDateTime that is null
+    final ZonedDateTimeObj obj = gson.fromJson("{\"time\": null}", ZonedDateTimeObj.class);
+
+    assertThat(obj.time).isNull();
+  }
+
   /** GSON cannot work with local classes */
-  private static final class Obj {
+  private static final class LocalDateTimeObj {
     LocalDateTime time;
   }
+
+  /** GSON cannot work with local classes */
+  private static final class ZonedDateTimeObj {
+    ZonedDateTime time;
+  }
 }
```

Comment:
I’m not sure that we use it anywhere, but should we also test the writing of `LocalDateTime` and `ZonedDateTime` values? You did implement that behavior, so maybe it should be tested (along with writing `null` as well).

Reply:
I was thinking of changing the `write` code to throw `UnsupportedOperationException` so I didn't have to worry about it, but since this code is destined for the SDK where we very well may have a use case for serializing these types, I'll just add those tests instead.